src/jit/morph.cpp

```diff
@@ -15631,14 +15641,15 @@ void                Compiler::fgPromoteStructs()
                 if (varDsc->lvIsParam)
                 {
 #if FEATURE_MULTIREG_STRUCT_PROMOTE
-                    if  (varDsc->lvIsMultiRegArgOrRet)   // Is this argument variable holding a value passed in multiple registers?
+                   
+                    if (varDsc->lvIsMultiregStruct()        &&   // Is this a variable holding a value that is passed in multiple registers?
+                        (structPromotionInfo.fieldCnt != 2) &&   // An argument with two exactly two fields
+                        false)                                   // TODO-PERF - Disabled as this needs additional implementation:
+                                                                 //             it hits assert(lvFieldCnt==1) in lclvar.cpp line 4417
                     {
```

Comment:
It would be better to say that this hits `noway_assert(varDsc->lvFieldCnt == 1)` in `lvaAssignVirtualFrameOffsetToArg1()` rather than mentioning a line number.